controllers/chat/moveToAnon.js

```diff
@@ -0,0 +1,130 @@
+const {responseError, responseSuccess} = require('../../utils/Responses');
+const {CHANNEL_TYPE} = require('../../helpers/constants');
+const UsersFunction = require('../../databases/functions/users');
+
+/**
+ *
+ * @param {import("express").Request} req
+ * @param {import("express").Response} res
+ */
+const moveToAnon = async (req, res) => {
+  const {
+    targetUser,
+    oldChannelId,
+    anon_user_info_color_code,
+    anon_user_info_color_name,
+    anon_user_info_emoji_code,
+    anon_user_info_emoji_name
+  } = req.body;
+
+  let members = [targetUser];
+  if (!members.includes(req.userId)) members.push(req.userId);
+
+  const client = StreamChat.getInstance(process.env.API_KEY, process.env.SECRET);
+  try {
+    /**
+     * @type {import('stream-chat').OwnUserResponse}
+     */
+    const user = {
+      name: `Anonymous ${anon_user_info_emoji_name}`,
+      id: req.userId,
+      image: '',
+      username: `Anonymous ${anon_user_info_emoji_name}`
+    };
+    await client.connectUser(user, req.token);
+
+    if (client.user.name !== `Anonymous ${anon_user_info_emoji_name}`) {
+      await client.upsertUser({id: req.userId, name: `Anonymous ${anon_user_info_emoji_name}`});
+    }
+
+    const newChannel = client.channel('messaging', {members});
+    const createdChannel = await newChannel.create();
+
+    const targets = members.filter((member) => member !== req.userId);
+    await targets.map(async (target) => {
+      const userModel = await UsersFunction.findUserById(User, target);
+
+      if (userModel.is_anonymous) {
+        const checkChatAnonUserInfo = await ChatAnonUserInfo.findOne({
+          where: {
+            channel_id: oldChannelId,
+            my_anon_user_id: req.userId,
+            target_user_id: target
+          }
+        });
+
+        if (checkChatAnonUserInfo === null) {
+          await ChatAnonUserInfo.create({
+            channel_id: newChannel.id,
+            my_anon_user_id: req.userId,
+            target_user_id: target,
+            anon_user_info_color_code:
+              checkChatAnonUserInfo?.anon_user_info_color_code || anon_user_info_color_code,
+            anon_user_info_color_name:
+              checkChatAnonUserInfo?.anon_user_info_color_name || anon_user_info_color_name,
+            anon_user_info_emoji_code:
+              checkChatAnonUserInfo?.anon_user_info_emoji_code || anon_user_info_emoji_code,
+            anon_user_info_emoji_name:
+              checkChatAnonUserInfo?.anon_user_info_emoji_name || anon_user_info_emoji_name
+          });
+        }
+      }
+    });
+
+    try {
+      await newChannel.updatePartial({
+        set: {
+          channel_type: CHANNEL_TYPE.ANONYMOUS,
+          anon_user_info_color_code,
+          anon_user_info_color_name,
+          anon_user_info_emoji_code,
+          anon_user_info_emoji_name,
+          better_channel_member: newChannel.state.members
+        }
+      });
+    } catch (e) {
+      console.log(e);
+    }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

Error handling for the `newChannel.updatePartial()` call is done by logging the error to the console. It would be better to handle this error more gracefully, perhaps by sending a response back to the client indicating that something went wrong.



<!-- This is an auto-generated comment by OSS CodeRabbit -->